tuning/data/data_handlers.py

```diff
@@ -160,23 +160,27 @@ def tokenize_and_apply_input_masking(
 
 
 def __wrap_jinja_rendering_with_exception_handling(render_template: callable, **kwargs):
+    base_err = (
+        "Failed to rended provided chat/custom template on dataset."
+        + "Please check your dataset and the template. Likely failure cause - "
+    )
     try:
         return render_template(**kwargs)  # <-------- Actual function call
     except TemplateSyntaxError as e:
         raise ValueError(
-            f"Invalid template syntax in provided jinja template. {e}"
+            f"{base_err}Provided jinja template syntax is invalid {e}"
         ) from e
     except UndefinedError as e:
         raise KeyError(
-            f"The dataset does not contain the key used in the provided jinja template. {e}"
+            f"{base_err}Jinja template keys are not present in the dataset{e}"
         ) from e
     except SecurityError as e:
         raise ValueError(
-            f"Unsafe operation detected in the provided Jinja template. {e}"
+            f"{base_err}Unsafe operation detected in the Jinja template. {e}"
         ) from e
     except Exception as e:
         raise ValueError(
-            f"Error occurred while rendering the provided Jinja template. {e}"
+            f"{base_err}Error occurred while rendering the Jinja template. {e}"
```

Comment:
Isn't `Error occurred while rendering the Jinja template.` redundant when the base_err states the same.